### PR TITLE
owner:  add setting/getting the owner value info

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43352

Problem Summary:

### What is changed and how it works?
Add `SetOwnerOpValue` for `Manager`.
Add `GetOwnerOpValue` to the file of owner.go.
`GetOwnerInfo` is not quite the right, rename it `GetOwnerKey`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: owner/manager.go [hunk 1]
```diff
@@ -65,6 +69,25 @@ const (
 	keyOpDefaultTimeout = 5 * time.Second
 )
 
+// OpType is the owner key value operation type.
+type OpType byte
+
+// List operation of types.
+const (
+	OpNone              OpType = 0
+	OpGetUpgradingState OpType = 1
```
Comment (tangenta) [inline owner/manager.go:78]: Can we simplify it to "OpUpgradingCluster"?

File: owner/manager.go [hunk 2]
```diff
@@ -46,6 +48,8 @@ type Manager interface {
 	RetireOwner()
 	// GetOwnerID gets the owner ID.
 	GetOwnerID(ctx context.Context) (string, error)
+	// SetOwnerOpValue updates the owner op value.
+	SetOwnerOpValue(ctx context.Context, op OpType) error
```
[the remark this answers is not in the document]
Comment (zimulala) [reply owner/manager.go:52]: Yes, this PR is the implementation, the next PR call.

File: owner/manager.go (hunk 1 above)
Comment (zimulala) [reply owner/manager.go:78]: This operation only means that the owner has received upgrading information, we cannot say that the cluster has received this information or other information

File: owner/manager.go [hunk 3]
```diff
@@ -274,32 +303,97 @@ func (m *ownerManager) revokeSession(_ string, leaseID clientv3.LeaseID) {
...
+			}
+		}
+	})
+
+	resp, err := m.etcdCli.Txn(ctx).
+		If(clientv3.Compare(clientv3.ModRevision(ownerKey), "=", modRevision)).
+		Then(clientv3.OpPut(ownerKey, string(newOwnerVal))).
+		Commit()
+	logutil.BgLogger().Info("set owner op value", zap.String("owner key", ownerKey), zap.ByteString("ownerID", ownerID),
+		zap.Stringer("old Op", op), zap.Stringer("op", op), zap.Bool("isSuc", resp.Succeeded), zap.Error(err))
```
[the remark this answers is not in the document]
Comment (zimulala) [reply owner/manager.go:384]: "old op" is the operation before modification, and "op" is the operation of setting

[the remark this answers is not in the document]
Comment (tangenta) [reply owner/manager.go:384]:
```
zap.Stringer("old Op", op), zap.Stringer("op", op)
```
But the argument is the same.

[the remark this answers is not in the document]
Comment (zimulala) [reply owner/manager.go]: Sorry, I fixed it.

**Review by tangenta — APPROVED** [review]

File: owner/manager.go [hunk 4]
```diff
@@ -65,6 +69,25 @@ const (
 	keyOpDefaultTimeout = 5 * time.Second
 )
 
+// OpType is the owner key value operation type.
+type OpType byte
+
+// List operation of types.
+const (
+	OpNone              OpType = 0
```
Comment (hawkingrei) [inline owner/manager.go:77]: We can use ```iota``` here.

**Review by hawkingrei — APPROVED** [review]

File: owner/manager.go (hunk 4 above)
Comment (zimulala) [reply owner/manager.go:77]: I think it's a little bit clearer to just assign it

**Pull request merged**